src/command.ts

```diff
@@ -326,6 +347,65 @@ export class Command<C extends Context = Context> implements MiddlewareObj<C> {
     return this;
   }
 
+  /**
+   * Finds the matching command in the given context
+   *
+   * @example
+   * ```ts
+   * // ctx.msg.text = "/delete_123 something"
+   * const match = Command.findMatchingCommand(/delete_(.*)/, { prefix: "/", ignoreCase: true }, ctx)
+   * // match is { command: /delete_(.*)/, rest: ["something"], match: ["delete_123"] }
+   * ```
+   */
+  public static findMatchingCommand(
+    command: MaybeArray<string | RegExp>,
+    options: CommandOptions,
+    ctx: Context,
+  ): CommandMatch | null {
+    const { matchOnlyAtStart, prefix, targetedCommands } = options;
+
+    if (!ctx.has(":text")) return null;
+
+    if (matchOnlyAtStart && !ctx.msg.text.startsWith(prefix)) {
+      return null;
+    }
+
+    const commandNames = ensureArray(command);
+    const commands = ctx.msg.text.split(prefix).map((text) => ({ text }));
+
+    for (const { text } of commands) {
+      const [command, username] = text.split("@");
```

Comment:
That's not good. for example `/send_email user@mail`

Reply:
This made me completely rewrite the command matching logic lol
But it works now, plus I've added tests for the new `findMatchingCommand` method, including one with a `@` in the command.

Reply:
That's good! Thank you!